src/lib/race.ts

```diff
@@ -34,6 +34,8 @@ export async function race(
     typeof container === 'string' ? document.querySelector<HTMLElement>(container) : container;
   if (!root) throw new Error('Container element is not found.');
 
+  validateOptions(options);
```

Comment:
I prefer that we make it a pure function that returns a new object with the valid options, like this:
```ts
const validOptions = validateOptions(options);
```

This makes it free of side-effects, can be easily re-used and tested.

you may use a pattern like this:

```ts
export function validateOptions(options: Partial<Options>): Partial<Options> {
  // validation logic here
  const dataShapes = ['long', 'wide', 'auto'];

  return {
    ...(is(options.title, 'string') ? { title: options.title } : {}),
    ...(includes(dataShapes, options.dataShape) ? { dataShape: options.dataShape } : {}),
    // rest of properties
  };
}
```

so the helper functions (like `is()` and `includes()` in this case) return `true` if the option is valid. Then it is added to the returned object, otherwise, discarded.

Please see [this for example](https://github.com/live-codes/livecodes/blob/develop/src/livecodes/config/validate-config.ts) (and you may use some of the helper functions used there if you wish)